l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -0,0 +1,742 @@
+            require(amount == _amount, "3T"); // The token has non-standard transfer logic
+        }
+
+        if (!hyperbridgingEnabled[_chainId]) {
+            chainBalance[_chainId][_l1Token] += _amount;
+        }
+        // Note we don't save the deposited amount, as this is for the base token, which gets sent to the refundRecipient if the tx fails
+    }
+
+    /// @dev Transfers tokens from the depositor address to the smart contract address
+    /// @return The difference between the contract balance before and after the transferring of funds
+    function _depositFunds(address _from, address _token, uint256 _amount) internal returns (uint256) {
+        if (_token == l1WethAddress) {
+            // Deposit WETH tokens from the depositor address to the smart contract address
+            uint256 balanceBefore = address(this).balance;
+            IERC20(l1WethAddress).safeTransferFrom(_from, address(this), _amount);
+            // Unwrap WETH tokens (smart contract address receives the equivalent amount of ETH)
+            IWETH9(l1WethAddress).withdraw(_amount);
+            uint256 balanceAfter = address(this).balance;
+
+            return balanceAfter - balanceBefore;
+        } else {
+            IERC20 token = IERC20(_token);
+            uint256 balanceBefore = token.balanceOf(address(this));
+            token.safeTransferFrom(_from, address(this), _amount);
+            uint256 balanceAfter = token.balanceOf(address(this));
+
+            return balanceAfter - balanceBefore;
+        }
+    }
+
+    /// @notice used by requestL2TransactionTwoBridges in Bridgehub
+    /// specifies called chainId and caller, and requested transaction in _data.
+    /// currently we only support a single tx, depositing.
+    function bridgehubDeposit(
+        uint256 _chainId,
+        address _prevMsgSender,
+        uint256 _l2Value,
+        bytes calldata _data
+    ) external payable override onlyBridgehub returns (L2TransactionRequestTwoBridgesInner memory request) {
+        require(l2BridgeAddress[_chainId] != address(0), "EB b. n dep");
+
+        (address _l1Token, uint256 _withdrawAmount, address _l2Receiver) = abi.decode(
+            _data,
+            (address, uint256, address)
+        );
+        if (_withdrawAmount != 0) {
+            /// This breaks the _depositeFunds function, it returns 0, as our balance doesn't increase
+            /// This should not happen, this bridge only calls the Bridgehub if Eth is the baseToken or for wrapped base token deposits
+            require(_prevMsgSender != address(this), "EB calling itself");
+
+            uint256 withdrawAmount = _depositFunds(_prevMsgSender, _l1Token, _withdrawAmount);
+            require(withdrawAmount == _withdrawAmount, "5T"); // The token has non-standard transfer logic
+        }
+        uint256 amount;
+
+        if (bridgehub.baseToken(_chainId) != _l1Token) {
+            // we are depositing wrapped baseToken
```

Comment:
probably I misunderstand something, but this looks very weird, `bridgehub.baseToken(_chainId) != _l1Token` means that we are depositing any token that it is not the base token.

then we set the `amount` to be equal to `_l2Value`, which is actually the `msg.value` of the final L1->L2 message, while `msg.value` should always be represented in the base token

Reply:
True, it should be `==` and not `!=`, sorry